### PR TITLE
fix: Ignore externally initiated transactions

### PR DESCRIPTION
**Pull request by red-0ne**
### Human Summary

`tClient` subscribes to all transactions that have `tClient.signinAddr` as sender.

When transactions are initiated externally (eg. `make acc_initialize_pubkeys`), `tClient` will try to handle the transaction and remove it's error channel which does not exists.

This change moves the `txErrorChans` cleanup where checks that the TX was initiated internally are already made.

#### Original error
image

### AI Summary

reviewpad:summary

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [x] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

- [x] **Run all unit tests**: `make go_develop_and_test`
- [x] **Run E2E tests locally**: `make test_e2e`
- [ ] **Run E2E tests on DevNet**: Add the `devnet-test-e2e` label to the PR. This is VERY expensive, only do it after all the reviews are complete.

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [x] I have performed a self-review of my own code
- [x] I have commented my code, updated documentation and left TODOs throughout the codebase

### CODE REVIEW

**Review by bryanchriswhite — APPROVED** [review]
Nice - thanks for fixing this! :raised_hands:

**Comment by okdas** [comment]
@red-0ne can you please merge in main to see if that clears an issue with e2e tests? Thanks!

File: pkg/client/tx/client.go [hunk 1]
```diff
@@ -365,34 +365,6 @@ func (tClient *txClient) goSubscribeToOwnTxs(ctx context.Context) {
 		txHashHex := txHashBytesToNormalizedHex(comettypes.Tx(txResult.Tx).Hash())
 
 		tClient.txsMutex.Lock()
```
Comment (Olshansk) [inline pkg/client/tx/client.go:367]: NIT: Wdyt of renaming `tClient` to `txClient`?

I was confused what it was referring to at first.

File: pkg/client/tx/client.go [hunk 2]
```diff
@@ -404,6 +376,36 @@ func (tClient *txClient) goSubscribeToOwnTxs(ctx context.Context) {
 			if len(txErrorChans) == 0 {
 				delete(tClient.txTimeoutPool, timeoutHeight)
 			}
+
+			// Check for a corresponding error channel in the map.
+			// Only close and remove from tClient.txErrorChans if it has been initiated
```
Comment (Olshansk) [inline pkg/client/tx/client.go:381]: Make sure to update the name in the comments as well (if we choose to rename the variable`

File: pkg/client/tx/client.go [hunk 3]
```diff
@@ -404,6 +376,36 @@ func (tClient *txClient) goSubscribeToOwnTxs(ctx context.Context) {
...
+			// by tClient.
+			// Transactions that are initiated externally will not have an associated
```
Comment (Olshansk) [inline pkg/client/tx/client.go:383]: I don't fully understand why being inside the loop solves it. You explained the `what` but not the `why`

#PUC

File: pkg/client/tx/client.go [hunk 4]
```diff
@@ -404,6 +376,36 @@ func (tClient *txClient) goSubscribeToOwnTxs(ctx context.Context) {
...
+			// error channel in tClient.txErrorChans and would trigger a panic if not
+			// handled inside this loop.
+			txErrCh, ok := tClient.txErrorChans[txHashHex]
+			if !ok {
+				panic("Received tx event without an associated error channel.")
+			}
+
+			// TODO_INVESTIGATE: it seems like it may not be possible for the
```
Comment (Olshansk) [inline pkg/client/tx/client.go:391]: Can you please shorten & simplify this. Reading the comment made it unclear to me whether our code path is callining `Sync` or `Async` broadcasting.

**Review by Olshansk — CHANGES_REQUESTED** [review]

File: pkg/client/tx/client.go (hunk 4 above)
Comment (red-0ne) [reply pkg/client/tx/client.go]: Updated the comment. @bryanchriswhite, could you please check if the shortened comment is still a valid?

File: pkg/client/tx/client.go [hunk 5]
```diff
@@ -135,33 +135,33 @@ func NewTxClient(
 	deps depinject.Config,
 	opts ...client.TxClientOption,
 ) (_ client.TxClient, err error) {
-	tClient := &txClient{
+	txClient := &txClient{
```
Comment (Olshansk) [inline pkg/client/tx/client.go:138]: `txClient := &txClient{`

I wanted to check whether this causes compilation issues or not?

File: pkg/client/tx/client.go [hunk 6]
```diff
@@ -203,7 +203,7 @@ func NewTxClient(
 // the synchronous error. If the function completes successfully, it returns an
 // either.AsyncError populated with the error channel which will receive if the
 // transaction results in an asynchronous error or times out.
-func (tClient *txClient) SignAndBroadcast(
+func (txClient *txClient) SignAndBroadcast(
```
Comment (Olshansk) [inline pkg/client/tx/client.go:206]: I think I see why we named it `tClient` before, some maybe `txnClient` is better.

Does this compile?

**Review by Olshansk — CHANGES_REQUESTED** [review]
@red-0ne 

1. Does this compile?
2. E2E tests are failing
3. Please sync w/ main

File: pkg/client/tx/client.go (hunk 5 above)
Comment (red-0ne) [reply pkg/client/tx/client.go:138]: Yes it compiles, golang differenciates the variable from the type.

**Comment by red-0ne** [comment]
> @red-0ne
> 
> 1. Does this compile?
> 2. E2E tests are failing
> 3. Please sync w/ main

1. It compiles.
2. E2E failure was old, succeeds locally, will see with `main` merged
3. Done

**Review by Olshansk — APPROVED** [review]
💯

**Pull request merged**